app/components/tab_navigation_component.rb

```diff
@@ -8,8 +8,9 @@ def initialize(label:, routes:, **tag_options)
   end
 
   def is_current_path?(path)
-    request.path == URI.parse(path).path
-  rescue URI::InvalidURIError
+    recognized_path = Rails.application.routes.recognize_path(path)
+    [recognized_path, request].pluck(:controller, :action).uniq.one?
```

Comment:
This hazards toward being overly "clever" in a way which harms readability. Open to revising.

This is basically equivalent to the expanded form...

```suggestion
    request[:controller] == recognized_path[:controller] &&
      request[:action] == recognized_path[:action]
```

Reply:
Could also extract a method for some self-documenting code, but at that point we've probably lost any benefit of "simplifying"

```suggestion
    same_hash_values?([recognized_path, request], [:controller, :action])
  end

  def same_hash_values?(hashes, properties)
    hashes.pluck(*properties).uniq.one?  
  end
```

Reply:
How about:
```suggestion
    request.slice(:controller, :action) == recognized_path.slice(:controller, :action)
```

Reply:
@zachmargolis Oops, missed your comment 'til now. Yeah, that might be a reasonable compromise of conciseness and readability!

Reply:
@zachmargolis Unfortunately doesn't work as-is, since `request` isn't actually a hash, it's a `ActionDispatch::Request` (`ActionDispatch::TestRequest` in spec).

Reply:
ah dang. Thanks for trying, I do agree with the simpler approach that you landed on